src/main/java/io/managed/services/test/client/kafkamgmt/KafkaMgmtApiUtils.java

```diff
@@ -195,6 +206,46 @@ public static void deleteKafkaByNameIfExists(KafkaMgmtApi api, String name) thro
             LOGGER.info("kafka instance '{}' not found", name);
         }
     }
+    
+    /**
+     * Delete all the Kafka Instances if it exists and if the SKIP_KAFKA_TEARDOWN env is set to false.
+     *
+     * @param api  KafkaMgmtApi
+     * @throws ApiGenericException, KafkaNotDeletedException
+     */
+    public static void cleanAllKafkaInstances(KafkaMgmtApi api) throws ApiGenericException, KafkaNotDeletedException {
+        if (Environment.SKIP_KAFKA_TEARDOWN) {
+            LOGGER.warn("skip kafka instance clean up");
+            return;
+        }
+        deleteAllKafkas(api);
+    }
+
+    /**
+     * Delete all Kafka Instances
+     *
+     * @param api  KafkaMgmtApi
+     * @throws ApiGenericException, KafkaNotDeletedException
+     */
+    public static void deleteAllKafkas(KafkaMgmtApi api) throws ApiGenericException, KafkaNotDeletedException {
```

Comment:
the drawback of this method is that we need to somehow let caller know about which instances will be actaully deleted.
assume situation: 
user A from org α  wants to delete all his instances (`instance-1`, `instance-2`)
user B from org α  has `instance-3` here. 

this method will loop forever when any of these users call this method because they will be also able to list instance of other user, but wont be able to delete it.  

Solution here would be to change `getAnyKafka` method to first filter only kafka instances owned by the user who actually calls that method.  
Additionally, also log, or comment on that method that it actually delete kafka instances (owned by given user)

Reply:
resolved by using deletion by name/id